sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -260,8 +260,8 @@ trait JoinSelectionHelper {
       canBuildLocalHashMapBySize(right, conf) && muchSmaller(right, left)
     }
     getBuildSide(
-      canBuildLeft(joinType) && buildLeft,
-      canBuildRight(joinType) && buildRight,
+      canBuildShuffledHashJoinLeft(joinType) && buildLeft,
```

Comment:
@maropu - `autoBroadcastJoinThreshold` has nothing to do with `canBuildShuffledHashJoinLeft/Right` and `canBuildBroadcastLeft/Right`. I am just refactoring the decision related to join type here. `autoBroadcastJoinThreshold` is used in `canBroadcastBySize` and `canBuildLocalHashMapBySize` which I didn't touch. IMO it would be risky to introduce a new config for SHJ here as users would have existing pipelines to be carefully tuning `autoBroadcastJoinThreshold` to enable SHJ.

Reply:
Ah, I see. That' the existing logic...